mailbox/plugin/quota-mailing-cassandra/src/test/java/org/apache/james/mailbox/quota/cassandra/CassandraQuotaThresholdHistoryStoreExtension.java

```diff
@@ -0,0 +1,80 @@
+
+package org.apache.james.mailbox.quota.cassandra;
+
+import org.apache.james.backends.cassandra.CassandraCluster;
+import org.apache.james.backends.cassandra.DockerCassandraExtension;
+import org.apache.james.backends.cassandra.DockerCassandraExtension.DockerCassandra;
+import org.apache.james.backends.cassandra.utils.CassandraUtils;
+import org.apache.james.mailbox.quota.QuotaThresholdHistoryStore;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+
+public class CassandraQuotaThresholdHistoryStoreExtension implements BeforeAllCallback, AfterAllCallback, BeforeEachCallback, AfterEachCallback, ParameterResolver {
+    private final DockerCassandraExtension dockerCassandraExtension;
+    private CassandraCluster cassandra;
+    private DockerCassandra dockerCassandra;
+    private CountChangesDao countChangesDao;
+    private SizeChangesDao sizeChangesDao;
+
+    public CassandraQuotaThresholdHistoryStoreExtension() {
+        dockerCassandraExtension = new DockerCassandraExtension();
+    }
+
+    @Override
+    public void beforeAll(ExtensionContext context) throws Exception {
+        dockerCassandraExtension.beforeAll(context);
+        dockerCassandra = dockerCassandraExtension.getDockerCassandra();
+    }
+
+    @Override
+    public void afterAll(ExtensionContext context) throws Exception {
+        dockerCassandraExtension.afterAll(context);
+    }
+
+    @Override
+    public void beforeEach(ExtensionContext context) throws Exception {
+        cassandra = CassandraCluster.create(
+                new CassandraQuotaThresholdHistoryStoreModule(), dockerCassandra.getIp(), dockerCassandra.getBindingPort());
+        countChangesDao = new CountChangesDao(cassandra.getConf(), CassandraUtils.WITH_DEFAULT_CONFIGURATION);
+        sizeChangesDao = new SizeChangesDao(cassandra.getConf(), CassandraUtils.WITH_DEFAULT_CONFIGURATION);
+    }
+
+    @Override
+    public void afterEach(ExtensionContext context) throws Exception {
+        cassandra.close();
+    }
+
+    @Override
+    public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
+        return (parameterContext.getParameter().getType() == QuotaThresholdHistoryStore.class);
+    }
+
+    @Override
+    public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
+        return new CassandraQuotaThresholdHistoryStore(countChangesDao, sizeChangesDao);
```

Comment:
You may instantiate this in `beforeEach`.
I you need it to be injected multiple times in the same test (I don't know why; but maybe...)
then you will get multiple instances.